src/experiments.js

```
@@ -209,15 +206,59 @@ Test.goalHandlers = {};
  * @param <Function> callback to bind an event listener.
  * @param <String[]> array of allowed node types to bind listener to.
  */
-Test.registerGoal = ( name, callback, closest = [] ) => {
-	Test.goalHandlers[ name ] = {
+const registerGoalHandler = ( name, callback, closest = [] ) => {
+	goalHandlers[ name ] = {
 		callback,
 		closest: Array.isArray( closest ) ? closest : [ closest ],
 	};
 };
 
+/**
+ *
+ * @param {HTMLElement} node The target node to attach the event listener to.
+ * @param {*} record The function called by the event listener to record an event.
+ * @param {*} on The JS dvent name to listen on.
+ */
+const bindListener = ( node, record, on ) => node && node.addEventListener( on, event => {
+	typeof record === 'function' && record( event );
```

Should we throw an error if record isn't a function?